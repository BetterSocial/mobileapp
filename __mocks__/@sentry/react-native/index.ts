export const addBreadcrumb = jest.fn();
