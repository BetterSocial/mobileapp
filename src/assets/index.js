import homeSelected from './icon/HomeSelected';
import homeUnselected from './icon/HomeUnselected';
import feedSelected from './icon/FeedSelected';
import feedUnselected from './icon/FeedUnselected';
import newsSelected from './icon/NewsSelected';
import newsUnselected from './icon/NewsUnselected';
import profileSelected from './icon/ProfileSelected';
import profileUnselected from './icon/ProfileUnselected';
import AllowLeft from './arrow-left.png';

const homeTabs = {
  homeSelected,
  homeUnselected,
  feedSelected,
  feedUnselected,
  newsSelected,
  newsUnselected,
  profileSelected,
  profileUnselected,
  AllowLeft
};

export {homeTabs};
