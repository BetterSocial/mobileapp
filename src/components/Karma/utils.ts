export const renderCircleColor = (karma: number) => {
  switch (true) {
    case karma > 87.5:
      return ['#50BA57', '#C7EECA', '#50BA57'];
    case karma > 75:
      return ['#70BF21', '#B3E57F', '#70BF21'];
    case karma > 62.5:
      return ['#A0C527', '#D1EA82', '#A0C527'];
    case karma > 50:
      return ['#D3D20C', '#F3F370', '#D3D20C'];
    case karma > 37.5:
      return ['#E4B92E', '#FFE591', '#E4B92E'];
    case karma > 25:
      return ['#F4833C', '#F9BE9A', '#F4833C'];
    case karma > 12.5:
    default:
      return ['#DF001F', '#F5B0B9', '#DF001F'];
  }
};
