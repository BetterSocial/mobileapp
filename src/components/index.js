import Avatar from './Avatar';
import SingleSidedShadowBox from './SingleSidedShadowBox';
import Footer from './Footer/Footer';

export {Avatar, SingleSidedShadowBox, Footer};
