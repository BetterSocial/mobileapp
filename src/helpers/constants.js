const MAX_POLLING_ALLOWED = 4
const MIN_POLLING_ALLOWED = 2
const MAX_POLLING_CHARACTER_ALLOWED = 25
const DEFAULT_PROFILE_PIC_PATH = "https://res.cloudinary.com/hpjivutj2/image/upload/v1617245336/Frame_66_1_xgvszh.png"

export {
    MAX_POLLING_ALLOWED,
    MIN_POLLING_ALLOWED,
    MAX_POLLING_CHARACTER_ALLOWED,
    DEFAULT_PROFILE_PIC_PATH
}