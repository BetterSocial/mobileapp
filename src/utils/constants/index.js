const ANALYTICS_SHARE_POST_FEED_ID = 'btn_share';
const ANALYTICS_SHARE_POST_FEED_SCREEN = 'feed_screen_btn_share';
const ANALYTICS_SHARE_POST_PDP_SCREEN = 'pdp_screen_btn_share';
const ANALYTICS_SHARE_POST_PROFILE_ID = 'btn_share';
const ANALYTICS_SHARE_POST_PROFILE_SCREEN = 'profile_screen_btn_share';
const ANALYTICS_SHARE_POST_TOPIC_ID = 'btn_share';
const ANALYTICS_SHARE_POST_TOPIC_SCREEN = 'topic_screen_btn_share';
const CHANNEL_TYPE_PERSONAL = 0;
const CHANNEL_TYPE_GROUP = 1;
const CHANNEL_TYPE_GROUP_LOCATION = 2;
const CHANNEL_TYPE_TOPIC = 3;
const CHANNEL_TYPE_ANONYMOUS = 4;
const DEFAULT_PROFILE_PIC_PATH =
  'https://res.cloudinary.com/hpjivutj2/image/upload/v1680929851/default-profile-picture_vrmmdn.png';
const DEFAULT_TOPIC_PIC_PATH =
  'https://res.cloudinary.com/hpjivutj2/image/upload/v1707840597/qlundmevnhtlwx6vlglo.png';
const DEFAULT_TOPIC_PIC_PATH_ANON =
  'https://res.cloudinary.com/hpjivutj2/image/upload/v1636632905/vdg8solozeepgvzxyfbv.png';
const KEY_ACCESS_TOKEN = 'access_token';
const KEY_ANONYMOUS_TOKEN = 'anonymous_token';
const KEY_REFRESH_TOKEN = 'refresh_token';
const MAX_POLLING_ALLOWED = 4;
const MAX_POLLING_CHARACTER_ALLOWED = 25;
const MIN_POLLING_ALLOWED = 2;
const POST_TYPE_LINK = 2;
const POST_TYPE_POLL = 1;
const POST_TYPE_STANDARD = 0;
const POST_VERB_LINK = 'link';
const POST_VERB_POLL = 'poll';
const POST_VERB_STANDARD = 'tweet';
const FEED_CHAT_KEY = 'feed_chats';

const DISCOVERY_TAB_USERS = 0;
const DISCOVERY_TAB_TOPICS = 1;
const DISCOVERY_TAB_DOMAINS = 2;
const DISCOVERY_TAB_NEWS = 3;

const SOURCE_FEED_TAB = 'feed tab';
const SOURCE_PDP = 'PDP';
const SOURCE_MY_PROFILE = 'my profile';

const PRIVACY_PUBLIC = 'public';
const PRIVACY_PEOPLE_I_FOLLOW = 'people_i_follow';

const ENABLE_DEV_ONLY_FEATURE = true;

const PREFIX_TOPIC = 'topic_';

const POST_CHECK_FEED_NOT_FOUND = 1;
const POST_CHECK_AUTHOR_NOT_FOLLOWING = 2;
const POST_CHECK_FEED_EXPIRED = 3;
const POST_CHECK_AUTHOR_BLOCKED = 4;

const PERMISSION_STATUS_NOT_DEFINED = 0;
const PERMISSION_STATUS_BLOCKED = 1;
const PERMISSION_STATUS_ACCEPTED = 2;
const PERMISSION_STATUS_PENDING = 3;
const CHAT_SIGNED = 'SIGNED';
const CHAT_ANON = 'ANON';

const NavigationConstants = {
  CREATE_POST_SCREEN: 'CreatePost'
};

const POST_VERSION = 2;

const DELETED_MESSAGE_TEXT = 'This message has been deleted';
const CONTEXT_MENU_REPLY = 'Reply';
const CONTEXT_MENU_COPY = 'Copy Message';
const CONTEXT_MENU_DELETE = 'Delete Message';
const MESSAGE_TYPE_REGULAR = 'regular';
const MESSAGE_TYPE_REPLY = 'reply';
const MESSAGE_TYPE_REPLY_PROMPT = 'reply_prompt';
const MESSAGE_TYPE_DELETED = 'deleted';
const FEEDS_DIFF_TO_FETCH = 8;
const IS_ANONYMOUS = 1;
const IS_NOT_ANONYMOUS = 0;

const getPostType = (postTypeInNumber) => {
  switch (postTypeInNumber) {
    case POST_TYPE_LINK:
      return 'post_link';
    case POST_TYPE_POLL:
      return 'post_poll';
    case POST_TYPE_STANDARD:
      return 'post_standard';
    default:
      return '';
  }
};

export {
  getPostType,
  ANALYTICS_SHARE_POST_FEED_ID,
  ANALYTICS_SHARE_POST_FEED_SCREEN,
  ANALYTICS_SHARE_POST_PDP_SCREEN,
  ANALYTICS_SHARE_POST_PROFILE_ID,
  ANALYTICS_SHARE_POST_PROFILE_SCREEN,
  ANALYTICS_SHARE_POST_TOPIC_ID,
  ANALYTICS_SHARE_POST_TOPIC_SCREEN,
  CHANNEL_TYPE_ANONYMOUS,
  CHANNEL_TYPE_GROUP,
  CHANNEL_TYPE_GROUP_LOCATION,
  CHANNEL_TYPE_PERSONAL,
  CHANNEL_TYPE_TOPIC,
  CHAT_ANON,
  CHAT_SIGNED,
  CONTEXT_MENU_COPY,
  CONTEXT_MENU_DELETE,
  CONTEXT_MENU_REPLY,
  DEFAULT_PROFILE_PIC_PATH,
  DEFAULT_TOPIC_PIC_PATH,
  DEFAULT_TOPIC_PIC_PATH_ANON,
  DELETED_MESSAGE_TEXT,
  DISCOVERY_TAB_DOMAINS,
  DISCOVERY_TAB_NEWS,
  DISCOVERY_TAB_TOPICS,
  DISCOVERY_TAB_USERS,
  ENABLE_DEV_ONLY_FEATURE,
  FEEDS_DIFF_TO_FETCH,
  FEED_CHAT_KEY,
  KEY_ACCESS_TOKEN,
  KEY_ANONYMOUS_TOKEN,
  KEY_REFRESH_TOKEN,
  MAX_POLLING_ALLOWED,
  MAX_POLLING_CHARACTER_ALLOWED,
  MESSAGE_TYPE_DELETED,
  MESSAGE_TYPE_REGULAR,
  MESSAGE_TYPE_REPLY,
  MESSAGE_TYPE_REPLY_PROMPT,
  MIN_POLLING_ALLOWED,
  NavigationConstants,
  PERMISSION_STATUS_NOT_DEFINED,
  PERMISSION_STATUS_ACCEPTED,
  PERMISSION_STATUS_BLOCKED,
  PERMISSION_STATUS_PENDING,
  POST_CHECK_AUTHOR_BLOCKED,
  POST_CHECK_AUTHOR_NOT_FOLLOWING,
  POST_CHECK_FEED_EXPIRED,
  POST_CHECK_FEED_NOT_FOUND,
  POST_TYPE_LINK,
  POST_TYPE_POLL,
  POST_TYPE_STANDARD,
  POST_VERB_LINK,
  POST_VERB_POLL,
  POST_VERB_STANDARD,
  POST_VERSION,
  PREFIX_TOPIC,
  PRIVACY_PEOPLE_I_FOLLOW,
  PRIVACY_PUBLIC,
  SOURCE_FEED_TAB,
  SOURCE_MY_PROFILE,
  SOURCE_PDP,
  IS_ANONYMOUS,
  IS_NOT_ANONYMOUS
};
