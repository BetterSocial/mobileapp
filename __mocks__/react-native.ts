import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN