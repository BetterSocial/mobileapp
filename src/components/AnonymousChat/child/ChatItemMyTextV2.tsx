/* eslint-disable no-unexpected-multiline */
// eslint-disable-next-line no-use-before-define
import * as React from 'react';
import Animated, {withDelay, withSequence, withTiming} from 'react-native-reanimated';
import ContextMenu, {ContextMenuAction} from 'react-native-context-menu-view';
import {Swipeable} from 'react-native-gesture-handler';
import {Linking, Text, TouchableOpacity, View, ViewStyle} from 'react-native';
import FastImage from 'react-native-fast-image';
import {useNavigation, useNavigation} from '@react-navigation/core';

import ChatReplyView from './ChatReplyView';
import IconChatCheckMark from '../../../assets/icon/IconChatCheckMark';
import IconChatClockGrey from '../../../assets/icon/IconChatClockGrey';
import IconVideoPlay from '../../../assets/icon/IconVideoPlay';
import IconFile from '../../../assets/icon/IconFile';
import useMessageHook from '../../../hooks/screen/useMessageHook';
import {
  CONTEXT_MENU_COPY,
  CONTEXT_MENU_DELETE,
  CONTEXT_MENU_REPLY,
  MESSAGE_TYPE_DELETED,
  MESSAGE_TYPE_REPLY,
  MESSAGE_TYPE_REPLY_PROMPT
} from '../../../utils/constants';
import {ChatItemMyTextProps} from '../../../../types/component/AnonymousChat/BaseChatItem.types';
import {ChatStatus} from '../../../../types/database/schema/ChannelList.types';
import {MessageType} from '../../../../types/hooks/screens/useMessageHook.types';
import {ScrollContext} from '../../../hooks/screen/useChatScreenHook';
import {calculateTime} from '../../../utils/time';
import {colors} from '../../../utils/colors';
import {
  containerStyle,
  dotStyle,
  messageStyle,
  styles,
  textContainerStyle,
  textStyle
} from './ChatItemText.style';
import {replyIcon} from './ChatItemTargetText';
import {formatBytes} from '../../../utils/string/StringUtils';

const ChatItemMyTextV2 = ({
  username = 'Anonymous User',
  time = '4h',
  isContinuous = false,
  message = '',
  attachments = [],
  status = ChatStatus.PENDING,
  avatar,
  chatType,
  messageType,
  data
}: ChatItemMyTextProps) => {
  const navigation = useNavigation();
  const {
    setReplyPreview,
    onContextMenuPressed,
    onOpenMediaPreview,
    pulseAnimation,
    animatedBubbleStyle,
    animatedPulseStyle
  } = useMessageHook();
  const scrollContext = React.useContext(ScrollContext);
  const swipeableRef = React.useRef<Swipeable>(null);

  React.useEffect(() => {
    if (scrollContext?.selectedMessageId === data?.id) {
      pulseAnimation.value = withSequence(
        withDelay(300, withTiming(1.1, {duration: 200})),
        withTiming(1, {duration: 200})
      );
      scrollContext?.setSelectedMessageId(null);
    }
  }, [scrollContext?.selectedMessageId]);

  const isReply = messageType === MESSAGE_TYPE_REPLY;
  const isReplyPrompt = messageType === MESSAGE_TYPE_REPLY_PROMPT;
  const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;

  const renderIcon = React.useCallback(() => {
    if (status === ChatStatus.PENDING)
      return (
        <View style={styles.iconNewLine}>
          <IconChatClockGrey color={colors.silver} width={12} height={12} />
        </View>
      );

    return (
      <View style={styles.iconNewLine}>
        <IconChatCheckMark color={colors.silver} width={12} height={12} />
      </View>
    );
  }, [status]);

  const contextMenuActions: ContextMenuAction[] = [
    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'}
  ];

  if (messageType !== MESSAGE_TYPE_DELETED) {
    contextMenuActions.push({title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'});
    contextMenuActions.push({title: CONTEXT_MENU_DELETE, systemIcon: 'trash', destructive: true});
  }

  const renderAvatar = React.useCallback(() => {
    if (!isShowUserInfo) return <View style={styles.avatar} />;
    return <View style={styles.mlBuble}>{avatar}</View>;
  }, [isShowUserInfo, avatar]);

  const onSwipeToReply = React.useCallback(
    (direction: 'left' | 'right') => {
      if (direction === 'right') return;
      if (swipeableRef.current) swipeableRef.current?.close();
      setReplyPreview({
        id: data?.id ?? data?.message?.id,
        user: {username},
        message,
        message_type: messageType as MessageType,
        updated_at: time,
        chatType,
        attachments
      });
    },
    [data, username, time, message, messageType, chatType, attachments]
  );

  return (
    <Swipeable
      testID="swipeable"
      ref={swipeableRef}
      friction={3}
      overshootFriction={2}
      onSwipeableOpen={onSwipeToReply}
      renderLeftActions={replyIcon}>
      <View style={containerStyle(true, isReplyPrompt)}>
        <Animated.View
          style={[
            styles.wrapper,
            animatedBubbleStyle as ViewStyle,
            animatedPulseStyle as ViewStyle
          ]}>
          <ContextMenu
            previewBackgroundColor="transparent"
            style={{flex: 1}}
            actions={contextMenuActions}
            onPress={(e) => onContextMenuPressed(e, data, chatType)}
            disabled={attachments.length > 1}>
            <View style={styles.radius8}>
              <View style={textContainerStyle(true, chatType)}>
                {isShowUserInfo && (
                  <View style={styles.chatTitleContainer}>
                    <Text style={[styles.userText, textStyle(true)]}>{username}</Text>
                    <View style={dotStyle(true)} />
                    <Text testID="timestamp" style={[styles.timeText, textStyle(true)]}>
                      {calculateTime(time, true)}
                    </Text>
                  </View>
                )}
                {attachments.length > 0 && messageType !== 'deleted' && (
                  <View
                    style={[
                      styles.attachmentContainer,
                      attachments?.find((item) => item.type === 'file') ? {height: 'auto'} : {}
                    ]}>
                    {attachments
                      .filter((item, index) => index <= 3)
                      .map((item, index) =>
                        item.type === 'file' ? (
                          <TouchableOpacity
                            key={index}
                            style={{flex: 1}}
                            activeOpacity={1}
                            onPress={() => Linking.openURL(item.file_path)}>
                            <View style={styles.attachmentFileContainer}>
                              <View style={styles.attachmentFileContent}>
                                <Text style={styles.attachmentFileName}>{item.file_name}</Text>
                                <View>
                                  <Text style={styles.attachmentFileInfo}>
                                    {formatBytes(item.file_size)} â€¢{' '}
                                    {item.file_name
                                      ?.split('.')
                                      [item.file_name?.split('.')?.length - 1]?.toUpperCase()}
                                  </Text>
                                </View>
                              </View>
                              <View style={styles.attachmentFileIcon}>
                                <IconFile />
                              </View>
                            </View>
                          </TouchableOpacity>
                        ) : (
                          <TouchableOpacity
                            key={index}
                            style={{
                              width: `${
                                (attachments.length >= 3 && index > 0) || attachments.length >= 4
                                  ? 50
                                  : 100
                              }%`,
                              height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
                              position: 'relative',
                              overflow: 'hidden'
                            }}
                            activeOpacity={1}
                            onPress={
                              attachments.length > 0 && item.type !== 'gif'
                                ? () => onOpenMediaPreview(attachments, index, navigation)
                                : null
                            }>
                            {item.type !== 'file' && (
                              <FastImage
                                style={styles.image}
                                source={{
                                  uri: item.thumb_url
                                }}
                              />
                            )}
                            {attachments.length > 4 && index === 3 && (
                              <View style={styles.moreOverlay}>
                                <Text style={styles.moreText}>+{attachments.length - 4}</Text>
                              </View>
                            )}
                            {/* Video Play Icon */}
                            {item.video_path && (
                              <View style={styles.moreOverlay}>
                                <IconVideoPlay />
                              </View>
                            )}
                          </TouchableOpacity>
                        )
                      )}
                  </View>
                )}

                <ChatReplyView
                  type={chatType}
                  messageType={messageType}
                  replyData={data?.reply_data}
                />

                <View style={{flexDirection: 'row'}}>
                  {attachments.length <= 0 && (
                    <Text style={messageStyle(true, messageType)}>{message}</Text>
                  )}
                  {renderIcon()}
                </View>
              </View>
            </View>
          </ContextMenu>

          {renderAvatar()}
        </Animated.View>
      </View>
    </Swipeable>
  );
};

export default React.memo(ChatItemMyTextV2);
