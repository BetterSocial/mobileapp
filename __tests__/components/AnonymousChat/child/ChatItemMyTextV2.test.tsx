/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
/* eslint-disable react/display-name */
import * as React from 'react';
import FastImage from 'react-native-fast-image';
import {act, fireEvent, render} from '@testing-library/react-native';

import ChatItemMyTextV2 from '../../../../src/components/AnonymousChat/child/ChatItemMyTextV2';
import {Context} from '../../../../src/context';
import {DEFAULT_PROFILE_PIC_PATH} from '../../../../src/utils/constants';

jest.spyOn(React, 'useRef').mockReturnValue({current: {close: jest.fn()}});

const avatar = (
  <FastImage
    style={{height: 24, width: 24, borderRadius: 12}}
    source={{uri: DEFAULT_PROFILE_PIC_PATH}}
  />
);

let setReplyPreview;
let contextValue;

beforeEach(() => {
  setReplyPreview = jest.fn();
  contextValue = {
    chat: [{replyTarget: null}, setReplyPreview],
    profile: [{myProfile: {}}]
  };
});

afterEach(() => {
  jest.restoreAllMocks();
});

const renderChatItemMyTextV2 = (isContinous = false) =>
  render(
    <Context.Provider value={contextValue}>
      <ChatItemMyTextV2
        avatar={avatar}
        username="username"
        isContinuous={isContinous}
        message="message"
        time="time"
        chatType="SIGNED"
        messageType="regular"
        data={{}}
      />
    </Context.Provider>
  );

describe('TESTING ChatItemMyTextV2', () => {
  it('should call setReplyPreview when swiped', () => {
    const {getByTestId} = renderChatItemMyTextV2();
    act(() => {
      fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
    });
    expect(setReplyPreview).toHaveBeenCalled();
  });

  it('should display the username and time when isContinuous is false', () => {
    const {getByText, getByTestId} = renderChatItemMyTextV2();
    expect(getByText('username')).toBeTruthy();
    expect(getByTestId('timestamp')).toBeTruthy();
  });

  it('should not display the username and time when isContinuous is true', () => {
    const {queryByText} = renderChatItemMyTextV2(true);
    expect(queryByText('username')).toBeNull();
    expect(queryByText('time')).toBeNull();
  });

  it('should display the message', () => {
    const {getByText} = renderChatItemMyTextV2();
    expect(getByText('message')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderChatItemMyTextV2().toJSON();
    expect(tree).toMatchSnapshot();
  });
});
