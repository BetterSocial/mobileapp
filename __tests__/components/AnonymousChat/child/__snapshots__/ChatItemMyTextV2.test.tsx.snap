// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING ChatItemMyTextV2 should match snapshot 1`] = `
<View
  friction={3}
  onSwipeableOpen={[Function]}
  overshootFriction={2}
  renderLeftActions={[Function]}
  testID="swipeable"
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "marginVertical": 2.4000000000000004,
        },
        Object {
          "paddingLeft": 60,
          "paddingRight": 10,
        },
        false,
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          },
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <ContextMenu
        actions={
          Array [
            Object {
              "systemIcon": "arrow.turn.up.left",
              "title": "Reply",
            },
            Object {
              "systemIcon": "square.on.square",
              "title": "Copy Message",
            },
            Object {
              "destructive": true,
              "systemIcon": "trash",
              "title": "Delete Message",
            },
          ]
        }
        onPress={[Function]}
        previewBackgroundColor="transparent"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 8,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#F5F5F5",
                  "borderRadius": 8,
                  "flex": 1,
                  "paddingHorizontal": 4.800000000000001,
                  "paddingVertical": 3.2,
                },
                Object {
                  "backgroundColor": "#4782D7",
                },
                false,
                Object {
                  "borderTopEndRadius": 0,
                },
                Object {
                  "paddingBottom": 14,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 10.8,
                      "lineHeight": 19.36,
                    },
                    Array [
                      Object {
                        "color": "#ffffff",
                      },
                    ],
                  ]
                }
              >
                username
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "borderRadius": 2,
                      "height": 3,
                      "marginHorizontal": 4,
                      "width": 3,
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                    },
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "fontFamily": "Inter-ExtraLight",
                      "fontSize": 9,
                      "lineHeight": 12.19,
                    },
                    Array [
                      Object {
                        "color": "#ffffff",
                      },
                    ],
                  ]
                }
                testID="timestamp"
              >
                NaNw ago
              </Text>
            </View>
            <Text
              onTextLayout={[Function]}
              style={
                Array [
                  Object {
                    "fontFamily": "Inter-Regular",
                    "fontSize": 14.4,
                    "lineHeight": 19.36,
                    "marginBottom": 4,
                  },
                  Object {
                    "color": "#ffffff",
                  },
                  false,
                ]
              }
            >
              message
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "flex-end",
                  "bottom": 6,
                  "position": "absolute",
                  "right": 8,
                }
              }
            >
              <View />
            </View>
          </View>
        </View>
      </ContextMenu>
      <View
        style={
          Object {
            "marginLeft": 3.2,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 12,
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <FastImageView
            defaultSource={null}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://res.cloudinary.com/hpjivutj2/image/upload/v1680929851/default-profile-picture_vrmmdn.png",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
